### ISSUE TITLE
Error doing job: Task exception was never retrieved

### ISSUE THREAD

**Issue by iz3man** [bimmerconnected/bimmer_connected#161]
Hi 

For sure nothing serious, but it keeps poping up as error all the time.
My 520d 2016 now (after several month) at least works somehow, I can preheat the car, sound horn etc, but can't read mileage and gas etc. (which never worked).
I can live with that, but following errors show up app the time, and i wanted to know what i can do against them ...

Thanks

```
Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 424, in _async_add_entity
    await entity.async_update_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 284, in async_update_ha_state
    self._async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 328, in _async_write_ha_state
    attr.update(self.device_state_attributes or {})
  File "/usr/src/homeassistant/homeassistant/components/bmw_connected_drive/lock.py", line 64, in device_state_attributes
    "door_lock_state": vehicle_state.door_lock_state.value,
AttributeError: 'NoneType' object has no attribute 'value'
```
```
bmw_connected_drive:
  name_of_car:
    username: USERNAME_BMW_CONNECTED_DRIVE
    password: PASSWORD_BMW_CONNECTED_DRIVE
    region: one of  "rest_of_world"
```

Hassio 0.105.5

**Comment by gerard33** [comment]
Made a PR with a fix in HA -> https://github.com/home-assistant/core/pull/32074
Goal is to have this fix in HA 0.107.

**Comment by gerard33** [comment]
This is fixed in Home Assistant 0.107.
